lib/truffle/stringio.rb

```diff
@@ -567,7 +567,10 @@ def sysread(length=nil, buffer='')
   end
 
   alias_method :readpartial, :sysread
-  alias_method :read_nonblock, :sysread
+  
+  def read_nonblock(length, buffer='', **options)
```

Comment:
This could just be declared as a keyword argument like `, exception: true`.

Reply:
It seems to accept any arguments and ignore ones it doesn't know about - would that still work with explicit keywords?

```
irb(main):001:0> require 'stringio'
=> false
irb(main):002:0> StringIO.new('foo').read_nonblock(3, foo: 1)
=> "foo"
```

Reply:
Interesting, it seems MRI indeed doesn't check it. So this is fine.
Indeed, adding the explicit keyword argument would prevent passing unknown kwargs.